db/crud.go

```diff
@@ -1059,7 +1059,11 @@ func (db *DatabaseCollectionWithUser) Put(ctx context.Context, docid string, bod
 		return newDoc, newAttachments, false, nil, nil
 	})
 
-	return newRevID, doc, err
+	if err == nil {
+		currentVersion = SourceAndVersion{SourceID: doc.HLV.SourceID, Version: doc.HLV.Version}
+	}
+
+	return newRevID, currentVersion, doc, err
```

Comment:
If this value is computable from everything in `doc` do we need a new return value from this method?

Places that care about `currentVersion` could call a method like `doc.CurrentVersion()` that just returns `SourceAndVersion{SourceID: doc.HLV.SourceID, Version: doc.HLV.Version}` ?